### PR TITLE
Feature/check error message interpreter

### LINKED ISSUE #59: The interpreter test suite doesn't check the error messages

### PR DESCRIPTION
**Pull request by Giska**
@cloderic please review

Fix #59

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased](https://github.com/craft-ai/craft-ai-client-python/compare/v1.6.0...HEAD) ##
+- Change error messages text to fit interpreter messages.
```
Comment (cloderic) [inline CHANGELOG.md]: Not the right format + you should be a little more precise, perhaps mention and link to the interpreter test suite.

File: craftai/errors.py [hunk 2]
```diff
@@ -57,7 +57,8 @@ def __init__(self, message, obj="URL"):
 class CraftAiDecisionError(CraftAiError):
   """Raised when some issue is encountered when trying to find a decision"""
   def __init__(self, message):
-    self.message = "".join(("Error while taking a decision: ", message))
+    print("message", message)
```
Comment (cloderic) [inline craftai/errors.py]: to remove

File: craftai/errors.py [hunk 3]
```diff
@@ -57,7 +57,8 @@ def __init__(self, message, obj="URL"):
...
+    self.message = "".join(("Unable to take decision, ", message))
```
Comment (cloderic) [inline craftai/errors.py]: ? `+`s don't work ?

File: craftai/interpreter.py [hunk 4]
```diff
@@ -33,15 +33,30 @@ class Interpreter(object):
...
   def decide(tree, args):
+    errors = []
     bare_tree, configuration, _ = Interpreter._parse_tree(tree)
     if configuration != {}:
       state = args[0]
       time = None if len(args) == 1 else args[1]
-      context = Interpreter._rebuild_context(configuration, state, time)
+      context_result = Interpreter._rebuild_context(configuration, state, time)
+      context = context_result["context"]
+      # errors = context_result["errors"]
```
Comment (cloderic) [inline craftai/interpreter.py]: remove

File: craftai/interpreter.py [hunk 5]
```diff
@@ -33,15 +33,30 @@ class Interpreter(object):
...
     else:
       context = Interpreter.join_decide_args(args)
 
-    Interpreter._check_context(configuration, context)
+    errors = Interpreter._check_context(configuration, context)
+
+    print("errors", errors)
```
Comment (cloderic) [inline craftai/interpreter.py]: to remove

File: craftai/interpreter.py [hunk 6]
```diff
@@ -33,15 +33,30 @@ class Interpreter(object):
...
+
+    # deal with missing properties
+    if errors:
+      message = "Unable to take decision, the given context is not valid: " + errors.pop(0)
+
+      for error in errors:
+        message = message + ", " + error
```
Comment (cloderic) [inline craftai/interpreter.py]: you can use a `', '.join(...)` here

File: craftai/errors.py (hunk 3 above)
Comment (Giska) [reply craftai/errors.py]: all the errors have this kind of look

**Pull request merged**